src/app/cart/components/cart-list/cart-list.component.html

```diff
@@ -1,14 +1,21 @@
-<h1>Your cart</h1>
-<h5>Total qty of Cart: {{ cart.totalQty }} pcs.</h5>
-<h5>Total cost of Cart: {{ cart.totalCost | currency:'EUR' }}</h5>
-<hr>
- <app-cart-item 
-  *ngFor="let cartItem of cart.cartItems; trackBy: trackByCartItemId"
-  [cartItem] = cartItem
-  (increaseQty)="onIncreaseQty($event)"
-  (decreaseQty)="onDecreaseQty($event)"
-  (deleteItem)="onDeleteItem($event)"
-  >
-  </app-cart-item>
-<hr>
-<button (click)="onDeleteAllItems()">Clear Cart</button>
\ No newline at end of file
+<ng-container *ngIf="cart.cartItems.length; then cartTemplate; else cartIsEmpty"></ng-container>
```

Comment:
looks like a little bit overcomplicated, you can remove ng-template cartTemplate
```
<ng-container *ngIf="cart.cartItems.length; else cartIsEmpty">
     <h1>Your cart</h1>
    <h5>Total qty of Cart: {{ cart.totalQty }} pcs.</h5>
    <h5>Total cost of Cart: {{ cart.totalCost | currency:'EUR' }}</h5>
    <hr>
...
</ng-container>
```

Reply:
Done. 
Thanks for clarification!